sonic_platform_base/sonic_xcvr/api/public/cmis.py

```diff
@@ -975,7 +979,8 @@ def set_lpmode(self, lpmode):
                 self.xcvr_eeprom.write(consts.MODULE_LEVEL_CONTROL, lpmode_val)
                 time.sleep(1)
                 mstate = self.get_module_state()
-                return True if mstate == 'ModuleReady' else False
+                return True if mstate == 'ModuleReady' or mstate=='ModulePwrUp' else False
```

Comment:
ModulePwrUp is a transient state. Why do we need to check this state here?.

Reply:
Because we try to do retry, it seems need 3-4sec until status=True on ZR. Or can you advise?

Reply:
We must check the steady state of the module. i.e. We will have to wait till the module reaches "ModuleReady" state.

Reply:
I retest and check spend time for set_lpmode=on. It spend time from 0.24 to 0.4sec on 400G-ZR until it get status=True. For set off, it speed time about 55sec until get ModuleReady.

Reply:
@jostar-yang please try using [get_datapath_init_duration()](https://github.com/sonic-net/sonic-platform-common/blob/master/sonic_platform_base/sonic_xcvr/api/public/cmis.py#L681) timeout to know how much time module can takes to transition from LowPwr to ModuleReady

Reply:
Call self.get_datapath_init_duration() after self.get_lpmode()==True. AP,get_datapath_init_duration, return "5000". What is unit for the API return code?
root@sonic:~# sfputil lpmode on Ethernet248
Enabling low-power mode for port Ethernet248 ... spend set_lpmdoe=on ,use 0.248994
CC:get dp_reinit_time=5000.000000
OK

Reply:
@jostar-yang Thanks. Please use this timeout value to wait upto max timeout duration.

Reply:
Do you means use "dp_deinit_time=5000" to be timeout value? What is it's unit? Is unit ms or us?

Reply:
I Check sonic_xcvr/codes/public/cmis.py and CMIS SPEC. Unit seems is ms. "5000" is from "0111b" that mapping to "1 s <= Tstate < 5 s"